modules/appeals_api/app/models/concerns/appeals_api/central_mail_status.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module CentralMailStatus
+    extend ActiveSupport::Concern
+
+    include SentryLogging
+
+    class_methods do
+      def refresh_statuses_using_central_mail!(appeals)
+
+        return if appeals.empty?
+
+        response = CentralMail::Service.new.status(appeals.pluck(:id))
+
+        unless response.success?
+          appeals[0].log_message_to_sentry('Error getting status from Central Mail', :warning, status: response.status, body: response.body)
+          raise Common::Exceptions::BadGateway
+        end
+
+        central_mail_status_objects = parse_central_mail_response(response).select { |struct| struct.id.present? }
+        ActiveRecord::Base.transaction do
+          central_mail_status_objects.each do |obj|
+            appeals.find { |h| h.id == obj.id }
+                   .update_status_using_central_mail_status!(obj.status, obj.error_message)
+          end
+        end
+      end
+
+      def date_from_string(string)
```

Comment:
This method shouldn't be pulled out. It's not the responsibility of the CentralMail interaction, it's more likely something that would go along with PDF construction or header parsing.

Reply:
Yeah, I think you're right. This is unrelated to Central Mail.